pkg/es/client/client.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// ResponseError holds information about a request error
+type ResponseError struct {
+	// Error returned by the http client
+	Err error
+	// StatusCode is the http code returned by the server (if any)
+	StatusCode int
+	// Body is the bytes readed in the response (if any)
+	Body []byte
+}
+
+// Error returns the error string of the Err field
+func (r ResponseError) Error() string {
+	return r.Err.Error()
+}
+
+func (r ResponseError) PrefixMessage(message string) ResponseError {
```

Comment:
does this need to be a public API? It does not seem to be used outside of this pkg